intel-sgx/pcs/src/pckcrt.rs

```diff
@@ -743,6 +748,26 @@ where
     }
 }
 
+/// NOTE: This conversion is only correct if all PCK certs in the vec have the
+/// same CA chain.
+impl<V> TryFrom<Vec<PckCert<V>>> for PckCerts
+where
+    V: VerificationType,
+{
+    type Error = ASN1Error;
+
+    fn try_from(mut pcks: Vec<PckCert<V>>) -> Result<PckCerts, ASN1Error> {
+        let pck_data = pcks.iter().map(|pck| pck.as_pck_cert_body_item()).collect::<Result<Vec<_>, ASN1Error>>()?;
+        // NOTE: assuming that all PCK certs in the vec have the same CA chain,
+        // so we pick the ca_chain from the first one:
```

Comment:
So what happens when an intermediate CA changes (e.g., one expired, or got revoked)? The PckCerts won't have a valid certificate chain anymore? Its not clear that Azure would register there old platform configurations again, just to get new a PckCert for each one. We may need to change how `PckCerts` stores this chain internally, if we fully want to emulate the API.

Reply:
I think this is an issue with the PCK Certs API in general since the issuer chain is returned as a separate header and not specific to each PCK cert, see `SGX-PCK-Certificate-Issuer-Chain` header in https://api.portal.trustedservices.intel.com/content/documentation.html#pcs-certificates-v4-response

Reply:
Yes, but Intel could easily sign these Pck certs with a different intermediate CA if they want to. It's an issue now because we need to emulate that API. But maybe this is something for another PR.